src/client/websocket/WebSocketShard.js

```diff
@@ -393,6 +379,27 @@ class WebSocketShard extends EventEmitter {
     this.emit(ShardEvents.CLOSE, event);
   }
 
+  /**
+   * Manually emit close since the ws never received the close frame.
+   */
+  emitClose() {
+    /**
+     * Emitted when a shard's WebSocket closes.
+     * @private
+     * @event WebSocketShard#close
+     * @param {CloseEvent} event The received event
+     */
+    this.emit(ShardEvents.CLOSE, {
+      code: 1011,
+      reason: WSCodes[1011],
+      /**
+       * - `wasClean` property is set to false when the WebSocket connection did not close via the close handshake.
+       * i.e. by not receiving a valid Close frame from the server.
+       */
+      wasClean: false,
+    });
+  }
+
```

Comment:
```suggestion
  /**
   * Manually emit close since the ws never received the close frame.
   * @param {CloseEvent} event The received event
   */
  emitClose(event = {
      code: 1011,
      reason: WSCodes[1011],
      wasClean: false,
    }) {
    /**
     * Emitted when a shard's WebSocket closes.
     * @private
     * @event WebSocketShard#close
     * @param {CloseEvent} event The received event
     */
    this.emit(ShardEvents.CLOSE, event);
  }

```


Now change 
https://github.com/discordjs/discord.js/blob/1856ac28504aad1a690cd5553852601baeb939b4/src/client/websocket/WebSocketShard.js#L373...L379
to 
```js
this.emitClose(event);
```